analysis/src/python/data_analysis/template_errors/search.py

```diff
@@ -137,11 +136,10 @@ def search(submissions_path: str, steps_path: str, result_path: str, steps_with_
 
     df_submissions = read_df(submissions_path)
 
-    lang = 'java17'
     df_steps = read_df(steps_path)
     # Parsing code templates
-    df_steps[StepColumns.CODE_TEMPLATES.value] = \
-        df_steps[StepColumns.CODE_TEMPLATES.value].map(lambda x: ast.literal_eval(x)[lang].split(os.linesep))
+    df_steps[StepColumns.CODE_TEMPLATE.value] = \
```

Comment:
Everything was right here, as this field **code_templates** comes from API in the format of a map  {'java17': '<some code on java17>'}. You can check it [here](https://hyperskill.org/api/docs/#steps-list) (e.x. id=2132) and you will see:

```
"code_templates": {
    "java17": "class Main {
        public static void main(String[] args) {
            // put your code here
        }
    }"
}
```

Reply:
I understand the problem. Maybe you should add the method `parse_template_code` and implement both variants for template parsing because API and database data have different fields format, but we want to support all of them.